cmd/theatre-envconsul/main.go

```diff
@@ -133,6 +134,37 @@ func mainError(ctx context.Context, command string) (err error) {
 			}
 		}
 
+		var filePaths = environment{}
+		const pathTBD = ""
+
+		// Rewrite vault-file: envs to vault: envs. Store the paths to which they should
+		// be written to in filePaths
+		for key, value := range env {
+			if strings.HasPrefix(value, "vault-file:") {
+				// determine if we define a path at which to place the file
+				trimmed := strings.TrimSpace(
+					strings.TrimPrefix(value, "vault-file:"),
+				)
+				if len(trimmed) == 0 {
+					return errors.Wrap(
+						fmt.Errorf("empty vault-file env var: %v", value),
+						"failed parse vault-file env var")
+				}
+
+				split := strings.SplitN(trimmed, ":", 2)
+				switch len(split) {
+				case 2: // path and key
+					filePaths[key] = split[1]
+					env[key] = fmt.Sprintf("vault:%s", split[0])
+				case 1: // just key
+					filePaths[key] = pathTBD
+					env[key] = fmt.Sprintf("vault:%s", trimmed)
+				default:
+					// will not execute because of SplitN, N=2
```

Comment:
I think it's less confusing to just remove this `default` case, and move the comment above the `switch` and amend appropriately